hal/inc/cellular_hal.h

```diff
@@ -122,6 +124,7 @@ cellular_result_t cellular_credentials_set(const char* apn, const char* username
  */
 CellularCredentials* cellular_credentials_get(void* reserved);
 
+bool cellular_sim_ready(void);
```

Comment:
modular is merged to develop so go ahead and add those to the cellular dynalib in `hal/inc/hal_dynalib_cellular.h`